spec/javascripts/NewService/formElements/FormWrapper.spec.jsx

```diff
@@ -1,38 +1,51 @@
+// @flow
+
 import React from 'react'
-import Enzyme, {mount} from 'enzyme'
+import Enzyme, {shallow} from 'enzyme'
 import Adapter from 'enzyme-adapter-react-16'
 
 import {FormWrapper} from 'NewService/components/FormElements'
 import {HiddenServiceDiscoveryInput} from 'NewService/components/FormElements'
-import {CSRFToken} from 'utilities/utils'
 
 Enzyme.configure({adapter: new Adapter()})
 
+const submitText = 'Add API'
 const props = {
   id: 'form-id',
   formActionPath: 'my-path',
   hasHiddenServiceDiscoveryInput: true,
-  submitText: 'Add API'
+  submitText
 }
 
 it('should render itself', () => {
-  const wrapper = mount(<FormWrapper {...props}/>)
+  const wrapper = shallow(<FormWrapper {...props}/>)
   expect(wrapper.find('#form-id').exists()).toEqual(true)
 })
 
+it('should render an empty input', () => {
+  const wrapper = shallow(<FormWrapper {...props}/>)
+  const button = wrapper.find(`input[name="utf8"][type='hidden']`)
+  expect(button.exists()).toBe(true)
+})
+
 it('should render submit button with proper text', () => {
```

Comment:
Maybe also another test to assert submitting the form when clicking the button?

Reply:
How would we do that?

Reply:
you render the form within an `act` scope, then find the button and dispatch a click event.

Reply:
Sure, but what is to be expected after the button is clicked? There's no UI feedback, no callback, just a pure http request.

Reply:
Right, sorry, there's no React involved in handling the submission, it's just left to Rails controller. Next time I suggest if you already know there's an obvious case and the person reviewing your code missed that, present it clearly, instead of fishing for a wrong answer. A simple "It's Rails managed, probably I'm missing something?" or "I might be missing something, why would we want to test a Rails form submission?" or "I think it's not needed since it's left to the Rails controller" or simply a "Why?" instead of a "How?" would have just satisfied the inquire and avoid the thread.... including this patronizing comment ;)

Reply:
OK, my bad 🙈